### ISSUE TITLE
'Fossa test' giving 500 error

### ISSUE THREAD

**Issue by jesse-krantzler** [fossas/fossa-cli#126]
I am trying to run the 'fossa test' command from my CI, and I cannot get it to succeed.

After running:
`FOSSA_API_KEY=<API_KEY> fossa -p=project-name`
`FOSSA_API_KEY=<API_KEY> fossa test --debug`

The results of the first 'fossa' command are successful:
```
...
============================================================
	    View FOSSA Report:
	    https://app.fossa.io/projects/custom%2B3092%2Fproject-name%2F/refs/branch/master/1523889542328/browse/dependencies
	
============================================================
Uploading build results (2/2)...
```

And then for the 'fossa test' debugging results, I get
```
DEBUG config:config/init.go:122/New Configuration initialized: config.CLIConfig{APIKey:"<API_KEY>", Fetcher:"custom", Project:"project-name", Revision:"8cca6b2a39a949ddbd0ca81491f61627df432286", Endpoint:"https://app.fossa.io", Modules:[]module.Config{module.Config{Name:"app", Path:"pom.xml", Type:"mvn"}, module.Config{Name:"nodejs-module-name", Path:"angular/package.json", Type:"nodejs"}}, Debug:true, DefaultCmd:config.DefaultConfig{Build:false}, AnalyzeCmd:config.AnalyzeConfig{Output:false, AllowUnresolved:false}, BuildCmd:config.BuildConfig{Force:false}, TestCmd:config.TestConfig{Timeout:600000000000}, UploadCmd:config.UploadConfig{Locators:false, Data:""}, ReportCmd:config.ReportConfig{Type:""}, ConfigFilePath:".fossa.yml"}
DEBUG test:main/test.go:45/getBuild Making Builds API request to: "https://app.fossa.io/api/revisions/custom+project-name$8cca6b2a39a949ddbd0ca81491f61627df432286/build"
DEBUG common:main/common.go:33/makeAPIRequest Making API request "PUT" "https://app.fossa.io/api/revisions/custom+project-name$8cca6b2a39a949ddbd0ca81491f61627df432286/build" "<API_KEY>" ""
Waiting for analysis to complete...
Waiting for analysis to complete...
Waiting for analysis to complete...
Waiting for analysis to complete...
Waiting for analysis to complete... 
DEBUG common:main/common.go:63/makeAPIRequest Response body: "No provided project found"
CRITICAL test:main/test.go:166/testCmd Could not execute test: could not make FOSSA build request: bad server response (500): "No provided project found"
 ```

and my .fossa.yml file is:
```# Generated by FOSSA CLI (https://github.com/fossas/fossa-cli)
# Visit https://fossa.io to learn more
version: 1
cli:
  server: https://app.fossa.io
  project: project-name
  fetcher: custom
analyze:
  modules:
  - name: app
    path: pom.xml
    type: mvn
  - name: nodejs-module-name
    path: angular/package.json
    type: nodejs
```

My project is visible and scanned in the fossa.io console, but for some reason, I cannot get the 'test' command to finish from the fossa-cli tool. I also tried a couple variations where I manually set the revision and the project name to what I think they should be, but continue to get the same results. Anyone have any ideas?

Examples of the other things I tried:
```
// Test with revision set to what the git repo revision number is(seems to match default command)
FOSSA_API_KEY=<API_KEY> fossa test -r=8cca6b2a39a949ddbd0ca81491f61627df432286 -p=project-name --debug 
```
```
// Test with revision set to what the fossa.io console seems to say the revision number is 
FOSSA_API_KEY=<API_KEY> fossa test -r=1523889542328 -p=project-name --debug
```

**Comment by xizhao** [comment]
Ah, I believe this is due to the mismatch in behavior on our side for handling custom locators.

 1. When the `fossa` command uploads data to generate a report, we allow the user to specify a custom project key for custom builds

 2. We will automatically namespace your chosen project key for a custom build only (behind the scenes), to make sure you do not interfere with builds from other users

 3. When we call `fossa test`, we query for the raw project key off of combining your configurations fetcher/project settings.  For a custom build, this will always result in a 404

We are working on a fix for this either in a configuration patch or a new API for querying test results.

**Comment by ilikebits** [comment]
This should be fixed in `v0.7.0-beta.1`. Feel free to reopen if this problem still occurs.
